File: smartmall-frontend/src/services/orderService.js
import api from "./api";

export const orderService = {
	list: (userId) => api.get("/orders", { params: userId ? { userId } : {} }),
	get: (id) => api.get(`/orders/${id}`),
	create: (payload) => api.post("/orders", payload),
	updateStatus: (id, status) => api.patch(`/orders/${id}/status`, null, { params: { status } }),
};

export default orderService;

File: smartmall-backend/src/main/java/com/smartmall/service/OrderService.java
package com.smartmall.service;

import com.smartmall.dto.OrderDTO;
import com.smartmall.entity.Order;

import java.util.List;

public interface OrderService {
	Order create(OrderDTO orderDTO);

	List<Order> getAll();

	List<Order> getByUserId(Long userId);

	Order getById(Long id);

	Order updateStatus(Long id, String status);
}
